packages/canary-web-components/src/components/ic-date-picker/ic-date-picker.tsx

```diff
@@ -0,0 +1,1512 @@
+    );
+  };
+
+  private nextYearButton = (): void => {
+    return this.renderMonthYearNavButton(
+      "next-year-button",
+      false,
+      !this.isNextYearAllowed()
+    );
+  };
+
+  private isPrevYearAllowed = (): boolean => {
+    return this.isYearAllowed(this.yearInView - 1);
+  };
+
+  private isNextYearAllowed = (): boolean => {
+    return this.isYearAllowed(this.yearInView + 1);
+  };
+
+  private isYearAllowed = (yr: number): boolean => {
+    return yearInRange(yr, this.minDate, this.maxDate);
+  };
+
+  private getMonthView = (date: Date): Date[] => {
+    const start = getWeekStart(getMonthStart(date), this.startOfWeek);
+    const end = getWeekEnd(getMonthEnd(date), this.startOfWeek);
+
+    const days: Date[] = [];
+    let current = start;
+
+    while (!dateMatches(current, end)) {
+      days.push(current);
+      current = new Date(current);
+      current.setDate(current.getDate() + 1);
+    }
+
+    days.push(current);
+
+    return days;
+  };
+
+  private isCurrentMonth = (): boolean => {
+    const d = new Date();
+    return (
+      d.getFullYear() === this.yearInView && d.getMonth() === this.monthInView
+    );
+  };
+
+  private updateMonthInView = (): void => {
+    this.currMonthView = this.getMonthView(this.focussedDate);
+
+    this.focussedDay = this.focussedDate.getDate();
+    this.monthInView = this.focussedDate.getMonth();
+    this.yearInView = this.focussedDate.getFullYear();
+
+    if (this.dayPickerKeyboardNav) {
+      this.monthInViewUpdateHandled = true;
+      this.setAriaLiveRegionText(this.getMonthInViewText());
+      this.dayPickerKeyboardNav = false;
+    }
+  };
+
+  private handleSelectDay = (day: Date): void => {
+    this.setSelectedDate(day);
+    this.calendarOpen = false;
+    this.inputEl.setCalendarFocus();
+  };
+
+  private handleSelectMonth = (month: number): void => {
+    this.moveMonths(month - this.monthInView);
+    this.monthPickerVisible = false;
+    this.setMonthSelectedLiveRegionText();
+
+    setTimeout(() => this.monthButtonEl.setFocus(), FOCUS_TIMER);
+  };
+
+  private handleSelectYear = (year: number): void => {
+    const yrPos = this.decadeView.indexOf(year);
+    if (yrPos > 0 && yrPos < this.decadeView.length - 1) {
+      this.moveYears(year - this.yearInView);
+      this.focusDay = false;
+      this.yearPickerVisible = false;
+      const monthName = this.monthNames[this.monthInView];
+      this.setAriaLiveRegionText(
+        `${year} selected. ${monthName} ${year} currently in view.`
+      );
+      setTimeout(() => this.yearButtonEl.setFocus(), FOCUS_TIMER);
+    } else {
+      const moveYears = year - this.focussedYear > 0 ? 10 : -10;
+      this.updateFocussedYear(moveYears, this.yearButtonFocussed);
+      this.setAriaLiveRegionText(this.getDecadeInViewText());
+    }
+  };
+
+  private monthPickerKeyDownHandler = (ev: KeyboardEvent): void => {
+    let handled = true;
+    switch (ev.key) {
+      case "ArrowUp":
```

Comment:
Since ArrowUp and ArrowLeft do the same thing here (and ArrowDown and ArrowRight) can the cases be put together like 
```
case "ArrowUp":
case "ArrowLeft":
  this.updateFocussedMonth(-1)
  break;
```